### ISSUE TITLE
`find_essential` in float32 misses the true essential matrix on an exact 5-point sample (N = 5 only; N >= 6 is fine)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4884]
On a noise-free minimal sample of exactly five correspondences, `find_essential` in float32 returns candidates none of which is close to the true `E`: the nearest is `0.11` away in Frobenius norm on unit-norm matrices, where float64 on the same five points recovers it to `2e-5` and float32 on six or more of the same points recovers it to `2e-5`. The minimal case is the one the 5-point solver exists for, and it is what `RANSAC(model_type="essential")` feeds it on every iteration, in the input's dtype.

## Repro

```python
import numpy as np, torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import essential_from_Rt, find_essential, projection_from_KRt

D = torch.float64
K1 = torch.tensor([[800.0, 0, 320], [0, 760, 200], [0, 0, 1]], dtype=D)
K2 = torch.tensor([[700.0, 0, 300], [0, 740, 240], [0, 0, 1]], dtype=D)
R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]], dtype=D))
t = torch.tensor([[[0.5], [0.1], [0.02]]], dtype=D)
X = torch.tensor([[-0.0075, 0.5364, 4.4163], [-0.7359, -0.3852, 5.8596], [-0.0198, 0.7929, 5.4462], [0.2646, -0.3022, 5.4847],
                  [-0.9553, -0.6623, 5.0526], [0.0370, 0.3953, 4.4873], [-0.6779, -0.4355, 5.1692], [0.8304, -0.2058, 4.0663],
                  [-0.1612, 0.1058, 4.2774], [-0.9277, -0.6295, 4.4845], [-0.3898, 0.8640, 5.6309], [-0.4603, -0.6986, 5.5863]], dtype=D)
P1 = projection_from_KRt(K1[None], torch.eye(3, dtype=D)[None], torch.zeros(1, 3, 1, dtype=D))
P2 = projection_from_KRt(K2[None], R, t)
h = lambda p: torch.cat([p, torch.ones_like(p[..., :1])], -1)
def proj(P, X):
    x = (P @ h(X[None]).transpose(1, 2)).transpose(1, 2)[0]
    return x[:, :2] / x[:, 2:]
x1, x2 = proj(P1, X), proj(P2, X)
n1 = (h(x1) @ torch.linalg.inv(K1).T)[:, :2]; n2 = (h(x2) @ torch.linalg.inv(K2).T)[:, :2]   # normalised coordinates
E_true = essential_from_Rt(torch.eye(3, dtype=D)[None], torch.zeros(1, 3, 1, dtype=D), R, t)[0]; E_true = E_true / E_true.norm()
for N in (5, 6, 8, 12):
    for dt in (torch.float64, torch.float32):
        Es = find_essential(n1[None, :N].to(dt), n2[None, :N].to(dt))[0]
        real = Es[~torch.isnan(Es).flatten(1).any(1)].double()
        dist = min(min((e - E_true).norm().item(), (e + E_true).norm().item()) for e in real)
        print(f"N={N:2d} {str(dt):14s} real candidates {len(real):2d}  min ||E ± E_true|| = {dist:.2e}")
```

```
N= 5 torch.float64  real candidates  6  min ||E ± E_true|| = 2.11e-05
N= 5 torch.float32  real candidates  4  min ||E ± E_true|| = 1.11e-01
N= 6 torch.float64  real candidates  6  min ||E ± E_true|| = 1.88e-05
N= 6 torch.float32  real candidates  6  min ||E ± E_true|| = 1.93e-05
N= 8 torch.float64  real candidates  4  min ||E ± E_true|| = 2.36e-05
N= 8 torch.float32  real candidates  4  min ||E ± E_true|| = 2.64e-05
N=12 torch.float64  real candidates  4  min ||E ± E_true|| = 3.68e-06
N=12 torch.float32  real candidates  4  min ||E ± E_true|| = 4.15e-06
```

kornia `main` 7ddf731bd, torch 2.14.0 and 2.5.1 (same numbers), CPU, macOS arm64. Found while pinning the `find_essential` contract in the batch-8 conventions audit (the review of the pin caught that it only verified the truth at N = 12).

## Classification

Silent wrong result in the documented minimal case. With N = 5 the null space of the 5 x 9 system is exactly four-dimensional and the Nistér elimination runs on that basis; the float32 path loses the solution there while the over-determined N >= 6 path, which computes the same four-vector basis from a least-squares null space, does not. Where the precision goes (the eigh-based null space, the Gauss–Jordan elimination on the 10 x 20 system, or the companion eigenvalues) is for the fix to establish. No idiom of 1–9; candidate: a minimal solver that is only correct in float64. Related: #4847 (candidate selection), #4855 (gradients for N < 9).

## Disposition

Focused fix, welcome as a PR: run the elimination and companion solve in float64 internally for float32 input (as `find_homography_dlt` and `triangulate_points` already upcast their linear algebra), or find and fix the float32-lossy step. A `test_wart_*` pin in the batch-8 docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by ducha-aiki** [comment]
Platform data from the batch-8a CI (#4887), for whoever fixes this: on the same exact 5-point sample in float32 the nearest candidate to the true `E` is `1.18` (Frobenius, unit-norm matrices) on macOS arm64 and `0.0175` on GitHub's ubuntu x86 runner, both on torch 2.14.0; float64 gives `2e-5` on both. So the size of the miss depends on the BLAS build, but the miss itself does not: float32 stays two to five orders of magnitude off the float64 result on both. The wart pin in #4887 therefore asserts `nearest > 1e-3`.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by ducha-aiki** [comment]
Follow-up from the next CI run of #4887 on the same ubuntu runner class: the float32 nearest-candidate distance was `0.0004` (torch 2.14.0) and `0.0003` (torch 2.5.1) this time, against `0.0175` in the previous run, `1.18` on macOS arm64 and `2e-5` in float64 everywhere. The float32 miss therefore ranges over four orders of magnitude across BLAS builds and even between runs on GitHub's runners, so #4887 documents this issue without a wart pin (a bound that holds on one runner fails on the next); the reproducible arm64 case stays in the issue as the reference. A fix that solves the elimination in float64 internally makes the result platform-independent, which is testable.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
